packages/flutter_tools/lib/src/commands/build_ios_framework.dart

```diff
@@ -0,0 +1,386 @@
+      throwToolExit('--universal or --xcframework is required.');
+    }
+    final List<BuildMode> buildModes = <BuildMode>[];
+
+    if (argResults['debug']) {
+      buildModes.add(BuildMode.debug);
+    }
+
+    if (argResults['profile']) {
+      buildModes.add(BuildMode.profile);
+    }
+
+    if (argResults['release']) {
+      buildModes.add(BuildMode.release);
+    }
+
+    if (buildModes.isEmpty) {
+      throwToolExit('--debug or --profile or --release is required.');
+    }
+
+    final String outputArgument = argResults['output'];
+
+    if (outputArgument.isEmpty) {
+      throwToolExit('--output is required.');
+    }
+
+    final BuildableIOSApp iosProject = await applicationPackages.getPackageForPlatform(TargetPlatform.ios);
+
+    if (iosProject == null) {
+      throwToolExit("Module's iOS folder missing");
+    }
+
+    final Directory outputDirectory = fs.directory(fs.path.normalize(outputArgument));
+
+    if (outputDirectory.existsSync()) {
+      outputDirectory.deleteSync(recursive: true);
+    }
+
+    aotBuilder ??= AotBuilder();
+    bundleBuilder ??= BundleBuilder();
+
+    for (BuildMode mode in buildModes) {
+      printStatus('Building framework for $iosProject in ${getNameForBuildMode(mode)} mode...');
+      final String xcodeBuildConfiguration = toTitleCase(getNameForBuildMode(mode));
+      final Directory modeDirectory = outputDirectory.childDirectory(xcodeBuildConfiguration);
+      final Directory iPhoneModeDirectory = modeDirectory.childDirectory('iphoneos');
+      final Directory iPhoneSimulatorDirectory = modeDirectory.childDirectory('iphonesimulator');
+
+      // Copy Flutter.framework.
+      await _produceFlutterFramework(outputDirectory, mode, iPhoneModeDirectory, iPhoneSimulatorDirectory, modeDirectory);
+
+      // Build aot, create module.framework and copy.
+      await _produceAppFramework(mode, iPhoneModeDirectory, modeDirectory);
+
+      // Build and copy plugins.
+      await processPodsIfNeeded(flutterProject.ios, getIosBuildDirectory(), mode);
+      if (hasPlugins(FlutterProject.current())) {
+        await _producePlugins(xcodeBuildConfiguration, iPhoneModeDirectory, iPhoneSimulatorDirectory, modeDirectory, outputDirectory);
+      }
+
+      final Status status = logger.startProgress(' └─Moving to ${fs.path.relative(modeDirectory.path)}', timeout: timeoutConfiguration.slowOperation);
+      if (iPhoneModeDirectory.existsSync()) {
+        iPhoneModeDirectory.deleteSync(recursive: true);
+      }
+
+      if (iPhoneSimulatorDirectory.existsSync()) {
+        iPhoneSimulatorDirectory.deleteSync(recursive: true);
+      }
+      status.stop();
+    }
+
+    printStatus('Frameworks written to ${outputDirectory.path}.');
+
+    return null;
+  }
+
+  Future<void> _produceFlutterFramework(Directory outputDirectory, BuildMode mode, Directory iPhoneModeDirectory, Directory iPhoneSimulatorDirectory, Directory modeDirectory) async {
```

Comment:
This needs error handling to prevent unhandled exceptions. Specifically:

- Failure to find expected files, 
- Failure to list/copy/write files due to permissions
- Process exceptions from compilation failures or misconfigured xcode
... et cetera

Reply:
I thought I agree when I was reading this but the more specific I get with implementing this, I question the premise of this request. 

Overall, we're not really actively doing much here besides shelling out commands to xcode. There's also lots of streamed output describing what shell step we're at. If it fails, we won't be able to describe much more than the current step (which is already printed) failed. 

i.e. I'm not sure we're historically done more good than harm attempting to parse platform tool messages and re-describing them by injecting wisdom while keeping the re-mapping from going stale https://github.com/flutter/flutter/blob/9093cf15f631b354c0fae4d726a8638b308a5e31/packages/flutter_tools/lib/src/ios/mac.dart#L583